File: 2000 98 & ME/QuickStart/howto/samples/Exceptions/Throw/throw.cs
using System;

class ExceptionTest{

	public static String SeeIfIntIsOne(int i)
	{
		String s = "You passed in 1";
		if (i == 1) 
			{
				return s;
			}
		else throw new ArgumentException("Your integer was not 1");
		// it should be noted that this exception does not utilize the runtime's resource
		// system which would make the text of the exception easily localizable.
		// For examples of using resources, see the Resources section of Quickstart.
	}

	public static void Main(String []args)
	{
		//This code shows how to catch an exception
		try 
		{
			Console.WriteLine("We're going to call our method with two args, one valid and one invalid");
			Console.WriteLine();
			Console.WriteLine("Trying to call the method with an argument of 1");
			Console.WriteLine(SeeIfIntIsOne(1));
			Console.WriteLine();
			Console.WriteLine("Trying to call the method with an argument of 2");
			Console.WriteLine(SeeIfIntIsOne(2));	
		}
		catch (Exception e)
		{
			Console.WriteLine("The following error occured:");
			Console.WriteLine(e.ToString());
		}
		finally 
		{
			Console.WriteLine();
			Console.WriteLine("Now we can continue");
		}

		Console.WriteLine ("\r\nPress Return to exit.");
		Console.Read();
	} // end Main

} // end class ExceptionTest

File: 2000 98 & ME/QuickStart/howto/samples/Exceptions/Throw/VB/throw.vb
Imports System

Public Class Throws

	Public Shared Function  SeeIfIntIsOne(i as Integer) As String
	
		Dim s As String 
		s = "You passed in 1"
		If i = 1
		   return s
		
		Else Throw New ArgumentException("Your integer was not 1")
		' it should be noted that this exception does not utilize the Runtime's resource
		' system which would make the text of the exception easily localizable.
		' For examples of using resources, see the Resources section of Quickstart.
		End If
	End Function

	Public Shared Sub Main ()
	
		'this code shows how to catch an exception
		Try 
		
			Console.WriteLine("We're going to call our mehod with two args, one valid and one invalid")
			Console.WriteLine()
			Console.WriteLine("Trying to call the method wih an argument of 1")
			Console.WriteLine(SeeIfIntIsOne(1))
			Console.WriteLine()
			Console.WriteLine("Trying to call the method with an argument of 2")
			Console.WriteLine(SeeIfIntIsOne(2))	
		
		Catch e As Exception 
		
			Console.WriteLine("The following error occured:")
			Console.WriteLine(e.ToString())
		
		Finally 
		
			Console.WriteLine()
			Console.WriteLine("Now we can Continue")
		End Try

		Console.WriteLine ()
		Console.WriteLine ("Press Return to exit.")
		Console.Read()
	End Sub ' end Main

End Class 'Throws
